File: Programmation/TD3/flexion-verbe.pl
#!/usr/bin/perl -w
use strict ;

my ( $radical, $inf, $verb, $i ) ;
my @terminaisons ;
my @pronoms = ( "je", "tu", "il/elle", "nous", "vous", "ils/elles" ) ;
my @terminaisons_1 = ( "e", "es", "e", "ons", "ez", "ent" ) ;
my @terminaisons_2 = ( "is", "is", "it", "issons", "issez", "issent" ) ;

sub Conjugate{
    print "Les formes du verbe \"$verb\" au présent de l'indicatif sont :\n" ;
    
    $inf = $verb ;
    $inf =~ s/.{2}$//;

    for ($i=0; $i<@pronoms; $i++)
    {
	print $pronoms[$i] . " " . $inf . $terminaisons[$i] . "\n" ;
    }
}

print "Entrez un verbe du 1er ou du 2eme groupe : " ;
chomp($verb=<STDIN>) ;

if ($verb =~ m/er$/){
    @terminaisons = @terminaisons_1 ;
    &Conjugate ;
}       
 
elsif ($verb =~ m/ir$/){
    @terminaisons = @terminaisons_2 ;
    &Conjugate ;
}

else{print "\"$verb\" n'est pas un verbe du premier ou du second groupe !\n"}

File: Programmation/TD3/flexion-verbe.rb
#!/usr/bin/ruby

$pronoms = [ "je", "tu", "il/elle", "nous", "vous", "ils/elles" ] ;
terminaisons_1 = [ "e", "es", "e", "ons", "ez", "ent" ] ;
terminaisons_2 = [ "is", "is", "it", "issons", "issez", "issent" ] ;


def conjugate(verb, terminaisons)
  puts "\nLes formes du verbe " + verb + " au présent de l'indicatif sont :\n"
  inf = verb.gsub(/.{2}$/, "")
  $pronoms.each_index{|x|
    print $pronoms[x] + " " + inf + terminaisons[x] + "\n"}
end

puts "Entrez un verbe du 1er ou 2nd groupe :\n"
verb = gets.chomp

if verb =~ /er$/
  conjugate(verb, terminaisons_1)
elsif
  verb =~ /ir$/
  conjugate(verb, terminaisons_2)
else
  puts "Ceci n'est pas un verbe du premier ou du second groupe !"
end
